arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -0,0 +1,1742 @@
+                    "description": "Gets or sets the Google API key."
+                }
+            },
+            "description": "Description of a NotificationHub GcmCredential."
+        },
+        "GcmCredential": {
+            "properties": {
+                "properties": {
+                    "$ref": "#/definitions/GcmCredentialProperties",
+                    "description": "Gets or sets properties of NotificationHub GcmCredential."
+                }
+            },
+            "description": "Description of a NotificationHub GcmCredential."
+        },
+        "MpnsCredentialProperties": {
+            "properties": {
+                "mpnsCertificate": {
+                    "type": "string",
+                    "description": "Gets or sets the MPNS certificate."
+                },
+                "certificateKey": {
+                    "type": "string",
+                    "description": "Gets or sets the certificate key for this credential."
+                },
+                "thumbprint": {
+                    "type": "string",
+                    "description": "Gets or sets the Mpns certificate Thumbprint"
+                }
+            },
+            "description": "Description of a NotificationHub MpnsCredential."
+        },
+        "MpnsCredential": {
+            "properties": {
+                "properties": {
+                    "$ref": "#/definitions/MpnsCredentialProperties",
+                    "description": "Gets or sets properties of NotificationHub MpnsCredential."
+                }
+            },
+            "description": "Description of a NotificationHub MpnsCredential."
+        },
+        "AdmCredentialProperties": {
+            "properties": {
+                "clientId": {
+                    "type": "string",
+                    "description": "Gets or sets the client identifier."
+                },
+                "clientSecret": {
+                    "type": "string",
+                    "description": "Gets or sets the credential secret access key."
+                },
+                "authTokenUrl": {
+                    "type": "string",
+                    "description": "Gets or sets the URL of the authorization token."
+                }
+            },
+            "description": "Description of a NotificationHub AdmCredential."
+        },
+        "AdmCredential": {
+            "properties": {
+                "properties": {
+                    "$ref": "#/definitions/AdmCredentialProperties",
+                    "description": "Gets or sets properties of NotificationHub AdmCredential."
+                }
+            },
+            "description": "Description of a NotificationHub AdmCredential."
+        },
+        "BaiduCredentialProperties": {
+            "properties": {
+                "baiduApiKey": {
+                    "type": "string",
+                    "description": "Get or Set Baidu Api Key."
+             
```

Comment:
why is NotificationHubResource not a child of "Resource". id, name, location, tags, type are properties already present in the "Resource" model. Please remove these properties from here and have an allOf on the Resource.